WordPress/gutenberg
# Site editor: preset custom color duplicates

Resolves https://github.com/WordPress/gutenberg/issues/39945

## What?

This PR ensures that we have unique default color palette item names and slugs.

## Why?

To prevent duplicates in the custom colors list.

## How?

This PR increments the id based on the highest available id in the elements array.

The result is as follows:

<img src="https://user-images.githubusercontent.com/6458278/166865637-75a4368f-59f9-4b94-bdfc-f39e94d0b64a.gif" width="400" />

## Testing Instructions

Get comfortable.

1. Open up the Site Editor
2. Head to Global Styles > Colors > Palette
3. Create some new solid custom colors by clicking on the `+` symbol.
4. Delete one of the colors (try from the start, then middle, then last) and create a new one. The name + id should increment from the last, _highest_ id.
5. When the list is empty, or there are no other `Color ${n}` names, you could always see `Color 1`
6. Click "Done" and make sure unedited colors are removed.
7. Switch from solid colours to gradients
8. Repeat the steps for gradients

**Note**: while testing you might notice the following error:

<img width="1339" alt="Screen Shot 2022-05-05 at 2 03 11 pm" src="https://user-images.githubusercontent.com/6458278/166865425-af1f6e07-ca58-43ff-b5fc-d96ad230403c.png">

This is because `CircularOptionPicker.Option` doesn't have unique keys when palettes have the same colors. 

I thought about preventing duplicate indices by concatenating the index in the PR, but I know that using indices as keys is discouraged. It's probably worth looking at in another PR where we could explore whether there is a more stable prop value we could use.
